Rename register() for container services

othercorey opened the pull request:
Suggest not using a core function name for optional feature.

If php had proper function overloading, then we could just do it that way.

othercorey committed:
Change register() to services() for containers

ndm2 commented:
This feels a little bit like an outlier, given the other method names in the class/interface. What about just `services`, or are there maybe plans to introduce `registerRoutes`, `registerCommands`, etc?

ADmad commented:
I too was thinking if we could just go with `services()`.

othercorey commented:
> I too was thinking if we could just go with `services()`.

That would fit.

ndm2 commented on src/Http/BaseApplication.php:
```diff
@@ -249,22 +249,22 @@ public function getContainer(): ContainerInterface
         }
-        $container = $this->register(new Container());
+        $container = $this->services(new Container());
         foreach ($this->plugins->with('register') as $plugin) {
```
I guess this should then be `services` too, with the `BasePlugin::$registerEnabled` property being renamed to `servicesEnabled` accordingly.

othercorey replied:
thats true

markstory commented:
I'm ok with the method being `services`. I used the name `register` as that is the name league uses in ServiceProviders. However, `services` is more aligned with the other method names in Application.

ADmad committed:
Change return type of ContainerApplicationInterface::services() to avoid.

The services() hooks should just mutate the provided instance,
no return value is necessary.

othercorey committed:
Update test function name

othercorey committed:
Merge pull request #15042 from cakephp/container-services

Change return type of ContainerApplicationInterface::services() to avoid.
